lib/prism/parse_result.rb

```diff
@@ -895,4 +909,46 @@ def initialize(locals, forwarding)
   def self.scope(locals: [], forwarding: [])
     Scope.new(locals, forwarding)
   end
+
+  # Given a Method, UnboundMethod or Proc, use its #source_location to parse the file and return a Prism node representing it.
+  # The returned node will either be a DefNode, LambdaNode or CallNode.
+  # Raises ArgumentError if it cannot be found for any reason.
+  # Only works on Ruby 4+ as it needs #source_location to contain column and end line information.
+  def self.node_for(callable)
```

Comment:
Could we have some tests for this?

For Ruby < 4, we could define singleton methods for this, as in:

```ruby
callable = method(:whatever)
if RUBY_VERSION < "4.0"
  def callable.source_location = [whatever]
end
```

that way we can get tests on this for all versions.